File: SereneEmployees/SereneEmployees.Web/Modules/ServerTypes/Personnel/EmployeeColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { EmployeeRow } from "./EmployeeRow";

export interface EmployeeColumns {
    Id: Column<EmployeeRow>;
    FullName: Column<EmployeeRow>;
    HireDate: Column<EmployeeRow>;
    Age: Column<EmployeeRow>;
    Salary: Column<EmployeeRow>;
}

export class EmployeeColumns extends ColumnsBase<EmployeeRow> {
    static readonly columnsKey = 'Personnel.Employee';
    static readonly Fields = fieldsProxy<EmployeeColumns>();
}

File: SereneEmployees/SereneEmployees.Web/Modules/Personnel/Employee/EmployeeColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace SereneEmployees.Personnel.Columns;

[ColumnsScript("Personnel.Employee")]
[BasedOnRow(typeof(EmployeeRow), CheckNames = true)]
public class EmployeeColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    [QuickFilter]
    public string FullName { get; set; }
    public DateTime HireDate { get; set; }
    [DisplayName("Age")]
    [QuickFilter]
    public Int32? Age { get; set; }
    public decimal Salary { get; set; }

}